modules/core/src/main/scala/trace4cats/package.scala

```diff
@@ -0,0 +1,30 @@
+import cats.data.Kleisli
+import cats.effect.kernel.Resource
+import trace4cats.model.{SpanKind, TraceHeaders}
+import trace4cats.{kernel => t4ckernel}
+
+import scala.annotation.unchecked.uncheckedVariance
+
+package object trace4cats {
+  type SpanName = String
```

Comment:
Unrelated: let's make this a value class instead of a type alias. But doesn't have to be now.

Reply:
This is something I considered when implementing this, I didn't want to burden the user with having to wrap every span name. We could have implicit conversions for strings to the value class, but then there is no obligation on the user to pass the _right_ string. It would be useful for internal library devs but an implicit conversion might create a bit of a footgun here.

Not sure how strongly you feel about this or if theres a really compelling reason for using a value class, I just wanted to lay out why this hasn't been done before.

Reply:
I don't feel super strongly about it, I mainly just dislike the illusory type safety provided by type aliases - what do you think about just using `String` explicitly?

Reply:
That's fair, it was added so that it could be easily identified in a tuple passed to `ResourceKleisli`, however this could be done using a case class as input rather than a tuple. I'll create an issue to do that and remove the type alias.

Reply:
#757